### DISCUSSION TITLE
textbox resizing/scaling

### DISCUSSION THREAD

**Discussion by VictoriaStefanikova** [fabricjs/fabric.js#7640] [Q&A] [answered]
Hi. Does anybody have experience with FabricJs and can support me on my problem? The issue is, that whenever I scale the Textbox, the text inside is also scaling. I want to disable scaling the text when scaling the textbox. After reading and implementing some stackoverflow suggestions, I came up with this: http://jsfiddle.net/vstefani/s7yfu93w/39/ but this is not exactly it, since it is not possible to scale both horizontaly and verticaly at the same time.

**Comment by jpomykala** [comment]
Calculate new width and height based on scaleX and scaleY properties, and then reset scaleX and scaleY to 1.

**Comment by ShaMan123** [comment, marked as answer]
The straightforward way to do this is to override the controls of textbox.
Take a look and see how the changeWidth/changeHeight controls are implemented and simply combine them for the other controls.
https://github.com/fabricjs/fabric.js/blob/6b920e79bab1413e41f6f33129654cd4a1c67760/src/controls.actions.js#L686
https://github.com/fabricjs/fabric.js/blob/2eabc92a3221dd628576b1bb029a5dc1156bdc06/src/mixins/default_controls.js#L101

duplicates #7377 #7541
